bluesky_queueserver/manager/tests/test_comms.py

```diff
@@ -223,7 +223,51 @@ def method_handler4(self, *, value=4):
     pc.stop()
 
 
-def test_PipeJsonRpcReceive_4_failing():
+def test_PipeJsonRpcReceive_4():
+    """
+    Test if all outdated unprocessed messages are deleted once the processing thread is started.
+    """
+
+    def method_handler3(*, value=3):
+        return value + 15
+
+    conn1, conn2 = multiprocessing.Pipe()
+    pc = PipeJsonRpcReceive(conn=conn2)
+    pc.add_method(method_handler3, "method3")
+
+    # The thread is not started yet, but we still send a message through the pipe.
+    #   This message is expected to be deleted once the thread starts.
+    request1a = [
+        format_jsonrpc_msg("method3", {"value": 5}),
+    ]
+    request1b = [
+        format_jsonrpc_msg("method3", {"value": 6}),
+    ]
+    conn1.send(json.dumps(request1a))
+    conn1.send(json.dumps(request1b))
+
+    # Start the processing thread. The messages that were already sent are expected to be ignored.
+    pc.start()
+
+    # Send another message. This message is expected to be processed.
+    request2 = [
+        format_jsonrpc_msg("method3", {"value": 7}),
+    ]
+    conn1.send(json.dumps(request2))
+
+    if conn1.poll(timeout=0.5):  # Set timeout large enough
+        response = conn1.recv()
+        response = json.loads(response)
+        assert len(response) == 1, "Unexpected number of response messages"
```

Comment:
I don't think it is needed. Actually I stopped putting messages in most cases, because it is additional work and messages clutter test output. `len(response)` is printed in case of failure anyway.

Reply:
OK, sounds good.